### PR TITLE
Refactor of language selection.

### PR DESCRIPTION
**Pull request by jorgegorka**
Some improvements to current implementation

### CODE REVIEW

File: cdx_core/spec/controllers/registrations_controller_spec.rb [hunk 1]
```diff
@@ -14,49 +14,55 @@
...
       get :edit, context: institution.uuid
+
       expect(response).to be_success
       expect(assigns(:locales)).to be_a(Array)
     end
 
     it "should update password" do
       old_pass = user.encrypted_password
-      params = {user: {password: '12345678', password_confirmation: "12345678"}}
+      params   = { user: {password: '12345678', password_confirmation: "12345678"}}
```
Comment (tripledub) [inline cdx_core/spec/controllers/registrations_controller_spec.rb]: If you are putting (correctly) whitespace inside left `{` then put one on the right side too. Never said anything before and not a big deal, but while we are talking about whitespace, I'm not a fan of lining up around the `=` as often makes lines unnecessarily long

File: cdx_core/spec/controllers/registrations_controller_spec.rb [hunk 2]
```diff
@@ -14,49 +14,55 @@
...
       post :update, params
+
       expect(user.reload.encrypted_password).not_to eq(old_pass)
     end
 
     it "should not update password if confirmation does not match" do
       old_pass = user.encrypted_password
-      params = {user: {password: '12345678', password_confirmation: "DIFFERENT"}}
+      params   = {user: {password: '12345678', password_confirmation: "DIFFERENT"}}
```
Comment (tripledub) [inline cdx_core/spec/controllers/registrations_controller_spec.rb]: `{ user: { password: '12345678', password_confirmation: 'DIFFERENT' } }`

File: cdx_core/app/controllers/application_controller.rb [hunk 3]
```diff
@@ -13,7 +13,7 @@ class ApplicationController < ActionController::Base
 
   def set_locale
```
Comment (tripledub) [inline cdx_core/app/controllers/application_controller.rb]: Can we now over-ride this method with a decorator?

Comment (jorgegorka) [inline cdx_core/app/controllers/application_controller.rb]: We don't need to override that method.

Comment (tripledub) [inline cdx_core/app/controllers/application_controller.rb]: Well maybe line #16 should be something like
```
I18n.locale = params[:language] || 'en'
```
But definitely not `vi`

Comment (jorgegorka) [inline cdx_core/app/controllers/application_controller.rb]: You are right.

**Pull request merged**